### PR TITLE
BLAIS5-3671: exclude "OSConfigAgent Error: unexpected end of JSON input" logs

### PR DESCRIPTION
**Pull request by Enigma-cloud**

### CODE REVIEW

File: lib/filters/osconfig_agent_filter.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from lib.log_processor import ProcessedLogEntry
+
+
+def osconfig_agent_filter(log_entry: ProcessedLogEntry):
+    entry_data = log_entry.data
+
+    # Skip "OSConfigAgent Error: unexpected end of JSON input" logs from GCE instances
+    if log_entry.platform == "gce_instance":
```
Comment (tomphp) [inline lib/filters/osconfig_agent_filter.py:8]: A [guard class](https://refactoring.guru/replace-nested-conditional-with-guard-clauses) could be used here to remove the nesting.

File: lib/send_alerts.py [hunk 2]
```diff
@@ -10,6 +10,18 @@
     CreateAppLogPayloadFromLogEntry,
 )
 from lib.log_processor import process_log_entry
+from lib.filters.osconfig_agent_filter import osconfig_agent_filter
+
+
+def log_entry_skipped(log_entry: ProcessedLogEntry):
+    print(log_entry)
```
Comment (tomphp) [inline lib/send_alerts.py:17]: This looks like a debug `print`, can it be removed?

File: tests/lib/filters/test_osconfig_agent_filter.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import datetime
+
+from lib.log_processor import ProcessedLogEntry
+from lib.filters.osconfig_agent_filter import osconfig_agent_filter
+
+
+def mock_log_entry(
```
Comment (tomphp) [inline tests/lib/filters/test_osconfig_agent_filter.py:7]: A more idiomatic approach would be to make this a `@pytest.fixture()`.
You could then use `dataclass.replace` to update values for each test.

See [this test](https://github.com/ONSdigital/blaise-slack-alerter/blob/main/tests/lib/log_processor/log_types/test_gce_instance.py#L39) for an example.

**Review by tomphp — CHANGES_REQUESTED** [review]

**Review by JAWilliamsONS — APPROVED** [review]
Gone through with Kris in person - looks really good and happy to approve
Obviously can't test in the real world so will merge and monitor if these alerts still come through
Unit tests are in place though 👍

**Review by JAWilliamsONS — APPROVED** [review]
changes made by Kris

**Pull request merged**